File: kotlin-root/gasnake/src/main/kotlin/com/example/snake/ai/ga/trymodel/Setting.kt
package com.example.snake.ai.ga.trymodel

import com.example.snake.ai.ActivationFunction
import com.example.snake.ai.ga.GaConfig
import com.example.snake.ai.ga.Options
import com.example.snake.ai.ga.SnakeBrainConfig

data class Setting(
    val evolveTimes: Int,
    val gaPlayerConfig: Options
)

val setting = com.example.snake.ai.ga.trymodel.Setting(
    evolveTimes = 2000,
    gaPlayerConfig = Options(
        worldWidth = 20,
        worldHeight = 20,
        snakeBrainConfig = SnakeBrainConfig(
            hiddenLayersLength = listOf(16, 8),
            hiddenLayerActivationFunction = ActivationFunction.LINEAR
        ),
        gaConfig = GaConfig(
            populationSize = 5000,
            surviveRate = 0.5,
            populationMutationRate = 0.2,
            geneMutationRate = 0.5,
            mutationAmount = 0.2,
            trialTimes = 1
        )
    )
)

File: node-root/snake-ai/scripts/setting.ts
import { ActivationFunction } from "../CalcUtils";
import type { Options } from "../GaModel";

interface Setting {
  evolveTimes: number;
  gaPlayerConfig: Options;
}

export const setting: Setting = {
  evolveTimes: 2000,
  gaPlayerConfig: {
    worldWidth: 20,
    worldHeight: 20,
    snakeBrainConfig: {
      hiddenLayersLength: [16, 8],
      hiddenLayerActivationFunction: ActivationFunction.LINEAR,
    },
    gaConfig: {
      populationSize: 2000,
      surviveRate: 0.5,
      populationMutationRate: 0.2,
      geneMutationRate: 0.5,
      mutationAmount: 0.2,
      trialTimes: 1,
    },
  },
};
